test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -1210,6 +1210,43 @@ def forward(self, x):
                 for key in n.meta:
                     self.assertEqual(n.meta[key], weight_meta[key])
 
+    def test_xnnpack_module_name_with_underscores(self) -> None:
+        """Test that if a module name has an underscore, we can still quantize it"""
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                # This module name has underscores, which can be part of a mangled
+                # name.
+                self.foo_bar = torch.nn.Linear(2, 2)
+
+            def forward(self, x):
+                return self.foo_bar(x)
+
+        quantizer = XNNPACKQuantizer()
+        # Set global to per-channel and then not per-channel for a specific submodule.
+        quantizer.set_global(get_symmetric_quantization_config(is_per_channel=True))
+        quantizer.set_module_name("foo_bar", get_symmetric_quantization_config(is_per_channel=False))
+        example_inputs = (torch.randn(2, 2),)
+        m = M().eval()
+        m = capture_pre_autograd_graph(m, example_inputs)
+        m = prepare_pt2e(m, quantizer)
+        # Check that the right quantization annotation is on the linear.
+        correct_qspec = QuantizationSpec(
+            dtype=torch.uint8,
+            quant_min=0,
+            quant_max=255,
+            qscheme=torch.per_tensor_symmetric,
+            is_dynamic=False,
+            observer_or_fake_quant_ctr=observer.HistogramObserver,
+        )
+        for n in m.graph.nodes:
+            if n.op == "call_function" and n.target == torch.ops.aten.linear.default:
+                self.assertIn("quantization_annotation", n.meta)
+                annotation = n.meta["quantization_annotation"]
+                self.assertIsInstance(annotation, QuantizationAnnotation)
+                self.assertEqual(annotation.input_qspec_map[n.args[0]], correct_qspec)
+                self.assertEqual(annotation.output_qspec, correct_qspec)
```

Comment:
also I think it will be better to test the prepared model inserts the correct activation_post_process (observer) module instead of testing the annotation

Reply:
maybe you can also simplify the test by setting `foo_bar` to be not quantized while the global setting is quantized

Reply:
Did this like you said (except the inverse, global is no quant, foo_bar is quantized)